vllm/model_executor/parallel_utils/communication_op.py

```diff
@@ -8,17 +8,20 @@
     get_tensor_model_parallel_world_size,
     get_tensor_model_parallel_group,
 )
+from vllm.model_executor.parallel_utils.custom_all_reduce import custom_all_reduce
 
 
 def tensor_model_parallel_all_reduce(input_: torch.Tensor) -> torch.Tensor:
     """All-reduce the input tensor across model parallel group.
 
-    NOTE: This operation is applied in-place on the input tensor.
+    NOTE: This operation may be applied in-place on the input tensor.
```

Comment:
can we put in the docstring when that will happen?

Reply:
yes